features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5XCrypto.cpp

```diff
@@ -0,0 +1,146 @@
+#include <algorithm>
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+#include <stdio.h>
+#include <string.h>
+
+
+#include "mbedtls/platform.h"
+#include "mbedtls/ecdh.h"
+#include "mbedtls/memory_buffer_alloc.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ecp.h"
+
+#include "platform/NonCopyable.h"
+#include "platform/CriticalSectionLock.h"
+#include "ble/BLETypes.h"
+#include "cmsis.h"
+#include "nRF5xCrypto.h"
+#include "platform/mbed_assert.h"
+
+namespace ble {
+namespace pal {
+namespace vendor {
+namespace nordic {
+
+LescCrypto::LescCrypto() : _initialized(false) {
+    mbedtls_entropy_init(&_entropy_context);
+    mbedtls_ecp_group_init(&_group);
+    int err = mbedtls_ecp_group_load(
+        &_group,
+        MBEDTLS_ECP_DP_SECP256R1
+    );
+    _initialized = err ? false : true;
+}
+
+LescCrypto::~LescCrypto() {
+    mbedtls_ecp_group_free(&_group);
+    mbedtls_entropy_free(&_entropy_context);
+}
+
+bool LescCrypto::generate_keys(
+    ArrayView<uint8_t> X,
+    ArrayView<uint8_t> Y,
+    ArrayView<uint8_t> secret
+) {
+    MBED_ASSERT(X.size() == public_key_coord_t::size());
+    MBED_ASSERT(Y.size() == public_key_coord_t::size());
+    MBED_ASSERT(secret.size() == public_key_coord_t::size());
+
+    mbedtls_mpi secret_key;
+    mbedtls_ecp_point public_keys;
+
+    mbedtls_mpi_init(&secret_key);
+    mbedtls_ecp_point_init(&public_keys);
+
+    int err = mbedtls_ecp_gen_keypair(
+        &_group,
+        &secret_key,
+        &public_keys,
+        mbedtls_entropy_func,
+        &_entropy_context
+    );
+
+    if (!err) {
+        store_mpi(secret, secret_key);
+        store_mpi(X, public_keys.X);
+        store_mpi(Y, public_keys.Y);
+    }
+
+    mbedtls_ecp_point_free(&public_keys);
+    mbedtls_mpi_free(&secret_key);
+
+    return err ? false : true;
+}
+
+bool LescCrypto::generate_shared_secret(
+    const ArrayView<const uint8_t>& peer_X,
+    const ArrayView<const uint8_t>& peer_Y,
+    const ArrayView<const uint8_t>& own_secret,
+    ArrayView<uint8_t> shared_secret
+) {
+    MBED_ASSERT(peer_X.size() == public_key_coord_t::size());
```

Comment:
since there are no longer types to enforce size requirements like this need to be documented in doxygen